### ISSUE TITLE
Uncaught TypeError: Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by Qinka** [atom-haskell/ide-haskell#67]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 0.207.0
**System**: Unknown Windows Version
**Thrown From**: [ide-haskell](https://github.com/atom-haskell/ide-haskell) package, v0.5.10

### Stack Trace

Uncaught TypeError: Cannot read property 'length' of undefined

```
At C:\Users\Qinka\AppData\Local\atom\app-0.207.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:409

TypeError: Cannot read property 'length' of undefined
  at TextBuffer.module.exports.TextBuffer.lineLengthForRow (C:\Users\Qinka\AppData\Local\atom\app-0.207.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:409:29)
  at TextBuffer.module.exports.TextBuffer.rangeForRow (C:\Users\Qinka\AppData\Local\atom\app-0.207.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:936:65)
  at TextEditor.module.exports.TextEditor.bufferRangeForBufferRow (C:\Users\Qinka\AppData\Local\atom\app-0.207.0\resources\app.asar\src\text-editor.js:677:26)
  at EditorControl.showExpressionType (C:\Users\Qinka\.atom\packages\ide-haskell\lib\editor-control.coffee:151:33)
  at C:\Users\Qinka\.atom\packages\ide-haskell\lib\editor-control.coffee:60:10

```

### Commands

```
     -0:10.3.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:09.1.0 atom-save-all:saveAll (atom-text-editor.editor.is-focused)
     -0:09.1.0 ide-haskell:prettify-file (atom-text-editor.editor.is-focused)
     -0:09.1.0 ide-haskell:check-file (atom-text-editor.editor.is-focused)
     -0:09.1.0 ide-haskell:lint-file (atom-text-editor.editor.is-focused)
     -0:07.8.0 core:undo (atom-text-editor.editor.is-focused)
     -0:04.1.0 atom-save-all:saveAll (atom-text-editor.editor.is-focused)
     -0:03.9.0 core:undo (atom-text-editor.editor.is-focused)
     -0:02.0 atom-save-all:saveAll (atom-text-editor.editor.is-focused)
     -0:02.0 ide-haskell:prettify-file (atom-text-editor.editor.is-focused)
     -0:01.9.0 ide-haskell:check-file (atom-text-editor.editor.is-focused)
     -0:01.9.0 ide-haskell:lint-file (atom-text-editor.editor.is-focused)
  4x -0:00.5.0 core:undo (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "destroyEmptyPanes": false,
    "themes": [
      "one-dark-ui",
      "atom-dark-syntax"
    ],
    "disabledPackages": [
      "autosave",
      "translator",
      "translator",
      "transformer",
      "transform"
    ]
  },
  "ide-haskell": {
    "onMouseHoverShow": "Info",
    "onSavePrettify": true,
    "useBackend": "haskell-ghc-mod",
    "activateStandalone": false
  }
}
```

### Installed Packages

```coffee
# User
atom-save-all, v0.2.0
atom-text-transformer, v0.7.0
autocomplete-haskell, v0.4.3
cabal, v0.0.15
haskell-ghc-mod, v0.7.4
haskell-grammar, v0.3.0
haskell-pointfree, v0.1.2
ide-haskell, v0.5.10
language-haskell, v1.0.0
language-powershell, v2.0.1
linter-hlint, v0.3.6
minimap, v4.9.4
save-commands, v0.6.2

# Dev
No dev packages
```

**Comment by lierdakil** [comment]
Could you provide more information on reproducing this? Backtrace seems nonsensical.

**Comment by simegeorge** [comment]
Just got this error when trying to prettify and Atom sent me here.  Although my error is the same, the trace is different so it may help diagnose...

---

[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 0.207.0
**System**: Microsoft Windows 8.1 Pro
**Thrown From**: [ide-haskell](https://github.com/atom-haskell/ide-haskell) package, v0.5.10


### Stack Trace

Uncaught TypeError: Cannot read property 'length' of undefined

```
At C:\Users\simon\AppData\Local\atom\app-0.207.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:409

TypeError: Cannot read property 'length' of undefined
    at TextBuffer.module.exports.TextBuffer.lineLengthForRow (C:\Users\simon\AppData\Local\atom\app-0.207.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:409:29)
    at TextBuffer.module.exports.TextBuffer.rangeForRow (C:\Users\simon\AppData\Local\atom\app-0.207.0\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:936:65)
    at TextEditor.module.exports.TextEditor.bufferRangeForBufferRow (C:\Users\simon\AppData\Local\atom\app-0.207.0\resources\app.asar\src\text-editor.js:677:26)
    at EditorControl.showExpressionType (C:\Users\simon\.atom\packages\ide-haskell\lib\editor-control.coffee:251:40)
    at C:\Users\simon\.atom\packages\ide-haskell\lib\editor-control.coffee:77:26
```

### Commands

```
     -1:50 core:save (atom-text-editor.editor.is-focused)
     -1:50 ide-haskell:check-file (atom-text-editor.editor.is-focused)
     -1:50 ide-haskell:lint-file (atom-text-editor.editor.is-focused)
     -1:45 editor:move-to-end-of-screen-line (atom-text-editor.editor.is-focused)
     -1:44.6.0 editor:newline (atom-text-editor.editor.is-focused)
     -1:41.3.0 core:save (atom-text-editor.editor.is-focused)
     -1:41.3.0 ide-haskell:check-file (atom-text-editor.editor.is-focused)
     -1:41.3.0 ide-haskell:lint-file (atom-text-editor.editor.is-focused)
  2x -1:18.8.0 core:move-up (atom-text-editor.editor.is-focused)
  2x -1:18.2.0 editor:move-to-first-character-of-line (atom-text-editor.editor.is-focused)
     -1:17.5.0 core:select-down (atom-text-editor.editor.is-focused)
     -1:17.2.0 core:delete (atom-text-editor.editor.is-focused)
     -1:16.2.0 core:save (atom-text-editor.editor.is-focused)
     -1:16.2.0 ide-haskell:check-file (atom-text-editor.editor.is-focused)
     -1:16.1.0 ide-haskell:lint-file (atom-text-editor.editor.is-focused)
     -0:03.8.0 ide-haskell:prettify-file (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-dark-ui",
      "atom-dark-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
haskell-ghc-mod, v0.7.5
ide-haskell, v0.5.10
language-haskell, v1.0.2

# Dev
No dev packages
```

**Comment by lierdakil** [comment]
Hmm. I think I know what's going on. Just published v0.5.11, which should help with this. I will close this in a month given there are no additional reports (since this bug rarely manifests and needs some very specific conditions to do so, testing for it would be relatively hard)

Thank you for reporting!

P.S. I don't understand what's with traces lately. Those sometimes show impossible call paths and nonsensical line numbers, yet when I manage to reproduce something, traces I get on *my* system make sense. Weird.
